ldap/servers/slapd/auditlog.c

```diff
@@ -118,20 +124,22 @@ write_auditfail_log_entry(Slapi_PBlock *pb)
 
     slapi_pblock_get(pb, SLAPI_OPERATION, &op);
     slapi_pblock_get(pb, SLAPI_TARGET_SDN, &sdn);
-
     slapi_pblock_get(pb, SLAPI_RESULT_CODE, &pbrc);
 
     switch (operation_get_type(op)) {
     case SLAPI_OPERATION_MODIFY:
         slapi_pblock_get(pb, SLAPI_MODIFY_MODS, &change);
+        slapi_pblock_get(pb, SLAPI_ENTRY_POST_OP, &entry);
         break;
     case SLAPI_OPERATION_ADD:
         slapi_pblock_get(pb, SLAPI_ADD_ENTRY, &change);
+        slapi_pblock_get(pb, SLAPI_ENTRY_POST_OP, &entry);
         break;
     case SLAPI_OPERATION_DELETE: {
         char *deleterDN = NULL;
         slapi_pblock_get(pb, SLAPI_REQUESTOR_DN, &deleterDN);
         change = deleterDN;
+        slapi_pblock_get(pb, SLAPI_ENTRY_PRE_OP, &entry);
```

Comment:
> Also, could you please clarify the design a bit?
> 
> I see that `auditfaillog` also is involved but the setting `nsslapd-auditlog-display-attrs` mentions only `auditlog`.
> 
> Does it mean that the setting propagates over auditfaillog all-together when set to auditlog?

In the design doc, the first sentence, it does say this is for both the audit and auditfail log.  I'll will try and make this more obvious that the config setting is for both audit logs.

Reply:
> If I try to restart after the test case, it crashes for me:
> 
> ```
> Stack trace of thread 59926:
> #0  0x00007ffa2eaed87f write_audit_file (libslapd.so.0 + 0xed87f)
> #1  0x00007ffa2eaee3e0 write_auditfail_log_entry (libslapd.so.0 + 0xee3e0)
> #2  0x00007ffa2eafea8c op_shared_delete (libslapd.so.0 + 0xfea8c)
> #3  0x00007ffa2eafed9d delete_internal_pb (libslapd.so.0 + 0xfed9d)
> #4  0x00007ffa2eb7fbc4 upgrade_server (libslapd.so.0 + 0x17fbc4)
> #5  0x0000562372ae9927 main (ns-slapd + 0x14927)
> #6  0x00007ffa2e629510 __libc_start_call_main (libc.so.6 + 0x29510)
> #7  0x00007ffa2e6295c9 __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x295c9)
> #8  0x0000562372aea945 _start (ns-slapd + 0x15945)
> ELF object binary architecture: AMD x86-64
> ```
> 
> But looking at the stack trace, I think, it happens just because `nsslapd-auditlog-logging-enabled` is on during init. And exactly delete operation, odd...

I can not reproduce the crash....

Reply:
Reproduced crash, you need auditfail log enabled :-)  Investigating...